rasterio/_io.pxd

```diff
@@ -19,6 +19,13 @@ cdef class BufferedDatasetWriterBase(DatasetWriterBase):
     pass
 
 
+cdef class WarpedVRTReaderBase(DatasetReaderBase):
```

Comment:
@sgillies I'm guessing this (and `rasterio.io.WarpedVRTReaderBase()`) exists to to leave the door open for a potential future `WarpedVRTWriterBase()`, but in my experience VRT's are extremely tricky to the point where I often just end up calling out to `gdalbuildvrt` through `subprocess` or relying on `gdalwarp -of VRT`.  I am hesitant to start thinking ahead to more generic Rasterio VRT tools and use cases until we have become more familiar with the VRT API and the complexities associated with exposing its functionality in Python.  If you are confident enough then 👍, but I am leery of introducing anything in this PR aside from directly exposing `AutoCreateWarpedVRT()`.

For reference, I think this is currently the best way to create a warped VRT on disk with Rasterio:

```python
import rasterio as rio
from rasterio.warp import reproject


with rio.open('tests/data/RGB.byte.tif') as src:
    dst_crs = 'EPSG:4326'
    profile = src.profile.copy()
    profile.update(
        crs=dst_crs,
        driver='VRT')
    with rio.open('out.vrt', 'w', **profile) as dst:
        for bidx in range(1, src.count + 1):
            reproject(
                rio.band(src, bidx),
                rio.band(dst, bidx),
                dst_crs=dst_crs)
```